### DISCUSSION TITLE
New Notification UX

### DISCUSSION THREAD

**Discussion by SteveAmor** [InfiniTimeOrg/InfiniTime#1844] [Ideas]
This is about the UX when new notifications are received.  Currently, when Notifications are enabled and you get a new notification, the motor vibrates and the new notification is displayed with a timeoutLine.  After 7 seconds (timeoutLength), the display returns to the previous screen.
This discussion is about the touch events *during* timeoutLenght which is managed at: https://github.com/InfiniTimeOrg/InfiniTime/blob/0f9f606b78ffbb93b1365156d8492bbf4ee4995d/src/displayapp/screens/Notifications.cpp#L151 
It appears to react to 3 touch events:
- Tap - timeoutLine is deleted before the screen times out and goes to sleep, remaining on the notification when next woken up.  When woken up, if there are previous notifications, you cannot swipe down to see them, you can only swipe right to dismiss or swipe up to go to previous screen.
- Swipe Right - dismisses the notification and returns to the previous screen.
- Swipe Up - takes you to the previous screen.

I can't see where the code allows Swipe Up during timeoutLenght but, aside from that, the UX doesn't seem right, having Tap.  I'd propose:
- Swipe Down - takes you to a previous notification if there was one. And you would then have to Swipe Up to go to the next notification.
- Swipe Right - dismiss notification and show previous notification if it existed, else display 0 notifications.
- Swipe Up - takes you to the previous screen.

All three would delete the timeoutLine.  All three would be consistent with swiping notifications.
At the very least, I think the Tap event should be removed, leaving only Swipe Right and Swipe Up.

**Comment by tituscmd** [comment]
I'm a bit late to the party on this discussion, but I recently noticed that you actually _can't_ swipe down to see the previous notification. I thought this was already a thing.
But I think the Tap event is nice, since sometimes you get a longer notification and by tapping it you kind of keep it in place to be able to keep reading it.

I might sit down later/in the following days and look what I can do about the Swipe Down gesture.

**Reply by SteveAmor** [reply]
Yes, agreed on the tap event being useful.

**Comment by SteveAmor** [comment]
Suggested improvement in #2469 
Maintained all existing functionality (including tap) and added swipeDown
